src/Compiler/Checking/PostInferenceChecks.fs

```diff
@@ -995,27 +997,27 @@ and CheckExprLinear (cenv: cenv) (env: env) expr (ctxt: PermitByRefExpr) (contf
         BindVal cenv env v
         LimitVal cenv v { limit with scope = if isByRef then limit.scope else env.returnScope }
         // tailcall
-        CheckExprLinear cenv env body ctxt contf 
+        CheckExprLinear cenv env body ctxt acc
 
     | LinearOpExpr (_op, tyargs, argsHead, argLast, m) ->
         CheckTypeInstNoByrefs cenv env m tyargs
         argsHead |> List.iter (CheckExprNoByrefs cenv env) 
         // tailcall
-        CheckExprLinear cenv env argLast PermitByRefExpr.No (fun _ -> contf NoLimit)
```

Comment:
(Still not convinced this is right.)

Reply:
Is this still unclear or are you happy with it?

Reply:
@Smaug123:
Is this comment resolved?
Is this PR now ready to be merged?

Reply:
Let's think step by step, he says, doing some prompt engineering on himself.

Suppose we have a `LinearMatchExpr[lim1](LinearOpExpr(LinearMatchExpr[lim2](something)))`.

Then in the new world, we compute `CombineTwoLimits lim1 (CombineTwoLimits NoLimit lim2)` and then throw it away to go into `CheckExpr something`.

In the old world, the result was `CheckExprLinear (fun limX -> id (CombineLimits [lim1 ; limX]))` applied to the `LinearOpExpr`; i.e. `CheckExprLinear (fun _ -> (fun limX -> id (CombineLimits [lim1 ; limX])) NoLimit)` applied to the innermost `LinearMatchExpr`; i.e. `CheckExprLinear (fun limY -> (fun _ -> (fun limX -> id (CombineLimits [lim1 ; limX])) NoLimit)) (CombineLimits [ lim2 ; limY ])` applied to the innermost `expr`; i.e. `(fun limY -> (fun _ -> (fun limX -> id (CombineLimits [lim1 ; limX])) NoLimit)) (CombineLimits [ lim2 ; limY ]) (CheckExpr[expr])`.
That reduces to `(fun _ -> lim1) (CombineLimits [ lim2 ; CheckExpr(expr) ])` which is `lim1`.

So I'm afraid this PR is not correct.